tests/functional/src/main/java/org/mule/functional/classloading/isolation/classification/MuleClassPathClassifier.java

```diff
@@ -0,0 +1,394 @@
+
+        return newArrayList(containerURLs);
+    }
+
+    private List<PluginUrlClassification> buildPluginsClassification(final ClassPathClassifierContext context, final Predicate<MavenArtifact> exclusion, final MavenArtifact compileArtifact, final MavenArtifactToClassPathURLResolver artifactToClassPathURLResolver, final File targetTestClassesFolder)
+    {
+        List<PluginUrlClassification> pluginClassifications = new ArrayList<>();
+        ClassPathScanningCandidateComponentProvider scanner = new ClassPathScanningCandidateComponentProvider(true);
+        scanner.addIncludeFilter(new AnnotationTypeFilter(Extension.class));
+        String extensionsBasePackage = getAnnotationAttributeFrom(context.getTestClass(), ArtifactClassLoaderRunnerConfig.class, "extensionBasePackage");
+        if (extensionsBasePackage != null && extensionsBasePackage.isEmpty())
+        {
+            throw new IllegalArgumentException("Base package for discovering extensions should be empty, it will take too much time to discover all the classes, please set a reasonable package for your extension annotate your tests with its base package");
+        }
+        Set<BeanDefinition> extensionsAnnotatedClasses = scanner.findCandidateComponents(extensionsBasePackage);
+
+        boolean isRootArtifactIdAnExtensionToo = false;
+        if (!extensionsAnnotatedClasses.isEmpty())
+        {
+            logger.debug("Extensions found, plugin class loaders would be created for each extension");
+            Set<String> extensionsAnnotatedClassesNoDups = extensionsAnnotatedClasses.stream().map(bd -> bd.getBeanClassName()).collect(Collectors.toSet());
+            for (String extensionClassName : extensionsAnnotatedClassesNoDups)
+            {
+                logger.debug("Classifying classpath for extension class: '{}'", extensionClassName);
+                Class extensionClass;
+                try
+                {
+                    extensionClass = forName(extensionClassName);
+                }
+                catch (ClassNotFoundException e)
+                {
+                    throw new IllegalArgumentException("Cannot create plugin/extension class loader classification due to extension class not found", e);
+                }
+
+                String extensionMavenArtifactId = getExtensionMavenArtifactId(extensionClass, context.getMavenMultiModuleArtifactMapping());
+                isRootArtifactIdAnExtensionToo |= compileArtifact.getArtifactId().equals(extensionMavenArtifactId);
+
+                pluginClassifications.add(extensionClassPathClassification(extensionClass, extensionMavenArtifactId, exclusion, artifactToClassPathURLResolver, context.getMavenDependencies(), compileArtifact, targetTestClassesFolder, context.getClassPathURLs()));
```

Comment:
Try to clean up this code. In particular, having a method extensionClassPathClassification taking eight parameters makes hard to understand what it does